packages/hdwallet-vultisig/src/ethereum.ts

```diff
@@ -0,0 +1,168 @@
+
+export function ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+  const slip44 = core.slip44ByCoin(msg.coin);
+  if (slip44 === undefined) return [];
+  return [
+    {
+      addressNList: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx, 0, 0],
+      hardenedPath: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx],
+      relPath: [0, 0],
+      description: "Vultisig",
+    },
+  ];
+}
+
+export async function ethSendTx(
+  msg: core.ETHSignTx,
+  vultisig: VultisigEvmProvider,
+  from: string
+): Promise<core.ETHTxHash | null> {
+  try {
+    const utxBase = {
+      from: from,
+      to: msg.to,
+      value: msg.value,
+      chainId: msg.chainId,
+      data: msg.data,
+      gasLimit: msg.gasLimit,
+    };
+
+    const utx = msg.maxFeePerGas
+      ? {
+          ...utxBase,
+          maxFeePerGas: msg.maxFeePerGas,
+          maxPriorityFeePerGas: msg.maxPriorityFeePerGas,
+        }
+      : { ...utxBase, gasPrice: msg.gasPrice };
+
+    const signedTx = await vultisig.request?.({
+      method: "eth_sendTransaction",
+      params: [utx],
+    });
+
+    return { hash: signedTx } as core.ETHTxHash;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+
+export async function ethSignMessage(
+  msg: core.ETHSignMessage,
+  vultisig: VultisigEvmProvider,
+  address: string
+): Promise<core.ETHSignedMessage | null> {
+  try {
+    if (!isHexString(msg.message)) throw new Error("data is not an hex string");
+    const signedMsg = await vultisig.request?.({
+      method: "personal_sign",
+      params: [msg.message, address],
+    });
+
+    return {
+      address: address,
+      signature: signedMsg,
+    } as ETHSignedMessage;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+
+export async function ethSignTypedData(
+  msg: core.ETHSignTypedData,
+  vultisig: VultisigEvmProvider,
+  address: string
+): Promise<core.ETHSignedMessage | null> {
+  try {
+    const signedMsg = await vultisig.request?.({
+      method: "eth_signTypedData_v4",
+      params: [address, JSON.stringify(msg.typedData)],
+    });
+
+    return {
+      address: address,
+      signature: signedMsg,
+    } as ETHSignedMessage;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+
+export async function ethGetAddress(vultisig: VultisigEvmProvider): Promise<core.Address | null> {
+  if (!(vultisig && vultisig.request)) {
+    return null;
+  }
+  try {
+    const ethAccounts = await vultisig.request({
+      method: "eth_accounts",
+    });
+    return ethAccounts[0];
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+export async function ethGetChainId(evmProvider: VultisigEvmProvider): Promise<number | null> {
+  try {
+    // chainId as hex string
+    const chainId: string = (await evmProvider?.request?.({ method: "eth_chainId" })) || "";
+    return parseInt(chainId, 16);
```

Comment:
nitpick - improve response handling
```suggestion
    const chainIdHex = await evmProvider.request?.({ method: "eth_chainId" });

    const chainId = parseInt(chainIdHex, 16);
    if (isNaN(chainId)) throw new Error(`Invalid eth_chainId response: ${chainIdHex}`);

    return chainId;
```